library/std/src/sys/unix/fs.rs

```diff
@@ -1126,6 +1126,19 @@ impl fmt::Debug for File {
             Some(PathBuf::from(OsString::from_vec(buf)))
         }
 
+        #[cfg(target_os = "freebsd")]
+        fn get_path(fd: c_int) -> Option<PathBuf> {
+            let info = Box::<libc::kinfo_file>::new_zeroed();
+            let mut info = unsafe { info.assume_init() };
+            (*info).kf_structsize = mem::size_of::<libc::kinfo_file>() as libc::c_int;
```

Comment:
You shouldn't need the `(*info)` here, and later too, since Box implements DerefMut and isn't a raw pointer here.